odoo/addons/base/models/res_users.py

```diff
@@ -489,11 +490,20 @@ def _check_action_id(self):
         # Prevent using reload actions.
         # We use sudo() because  "Access rights" admins can't read action models
         for user in self.sudo():
-            if user.action_id.type == "ir.actions.client":
-                action = self.env["ir.actions.client"].browse(user.action_id.id)  # magic
+            action_type = user.action_id.type
+            if not user.action_id.type:
+                continue
+            action = self.env[action_type].browse(user.action_id.id)  # magic
+            if action_type == "ir.actions.client":
                 if action.tag == "reload":
                     raise ValidationError(_('The "%s" action cannot be selected as home action.', action.name))
 
+            if action_type in ('ir.actions.client', 'ir.actions.act_window'):
+                user_context = self.with_user(user).context_get()
```

Comment:
I think this line assumes there is no context customization from the request.
The `user_context` doesn't have the `allowed_company_ids`. But the value is not really used by actions. So I think this assumption might be acceptable.